test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala

```diff
@@ -610,6 +610,64 @@ class OptimizerTest {
     }
   }
 
+  @Test def keepQualifierSideEffectsOfEliminatedField(): Unit = {
+    @noinline
+    class Foo(var x: Int)
+
+    val foo = new Foo(2)
+
+    var called = false
+
+    @noinline
+    def getFoo() = {
+      called = true
+      foo
+    }
+
+    getFoo().x = 1
+
+    assertTrue(called)
+  }
+
+  @Test def keepQualifierSideEffectsOfEliminatedFieldInline(): Unit = {
+    @inline
+    class Foo(var x: Int)
+
+    val foo = new Foo(2)
+
+    var called = false
+
+    @noinline
+    def getFoo() = {
+      called = true
+      foo
+    }
+
+    getFoo().x = 1
+
+    assertTrue(called)
+  }
+
+  @Test def keepQualifierSideEffectsOfEliminatedJSField(): Unit = {
```

Comment:
Note that this test doesn't really test anything, because we do not optimize JS classes (IIUC). Do you have any ideas of how I can trigger the relevant codepath?

Reply:
I was not able to get this to fail. I suspect that as soon as we hit the inline machinery, the local def machinery already lifts side-effects away, so we will not see qualifiers with side-effects anymore.

I propose we just leave the tests in the current state.